src/test/scala/chiselTests/MuxSpec.scala

```diff
@@ -43,14 +43,14 @@ class MuxLookupEnumTester extends BasicTester {
     case (e, i) =>
       e -> i.U
   }
-  assert(MuxLookup.fromEnum(TestEnum.a, 3.U)(mapping) === 0.U)
-  assert(MuxLookup.fromEnum(TestEnum.b, 3.U)(mapping) === 1.U)
-  assert(MuxLookup.fromEnum(TestEnum.c, 3.U)(mapping) === 2.U)
+  assert(MuxLookup(TestEnum.a, 3.U)(mapping) === 0.U)
+  assert(MuxLookup(TestEnum.b, 3.U)(mapping) === 1.U)
+  assert(MuxLookup(TestEnum.c, 3.U)(mapping) === 2.U)
 
   val incompleteMapping = Seq(TestEnum.a -> 0.U, TestEnum.c -> 2.U)
-  assert(MuxLookup.fromEnum(TestEnum.a, 3.U)(incompleteMapping) === 0.U)
-  assert(MuxLookup.fromEnum(TestEnum.b, 3.U)(incompleteMapping) === 3.U)
-  assert(MuxLookup.fromEnum(TestEnum.c, 3.U)(incompleteMapping) === 2.U)
+  assert(MuxLookup(TestEnum.a, 3.U)(incompleteMapping) === 0.U)
+  assert(MuxLookup(TestEnum.b, 3.U)(incompleteMapping) === 3.U)
+  assert(MuxLookup(TestEnum.c, 3.U)(incompleteMapping) === 2.U)
```

Comment:
Like a case where I define anotherEnum I should not be able to put AnotherEnum.c in for the default or any of the cases, I think.

Reply:
Agreed--we really need a testing flow for negative compile tests...

I have a prototype of the build flow to make this work, but we need some pithy way to express what we expect. Either a standalone `FileCheck`, a simpler custom similar thing, or perhaps Linux `expect`

Reply:
we could use a macro like the one shapeless uses https://github.com/milessabin/shapeless/blob/main/core/shared/src/main/scala/shapeless/test/typechecking.scala

Reply:
ScalaTest has something similar, but macros don't work super well because they aren't full separate invocations of the compiler and thus can't take into account differing compile options